### ISSUE TITLE
Locals are shared between clients

### ISSUE THREAD

**Issue by louie-github** [vxgmichel/aioconsole#69]
Hi! As of now, I'm using this code to start a remote server:

```
async def run_backdoor(bot=None):
        server_backdoor = await aioconsole.server.start_console_server(
            host="0.0.0.0", port=[PORT], locals={"bot": bot}
        )
````
I use `start_console_server` instead of `start_interactive_server` so I can pass `locals`.

However, when I use this server, when I connect to it using `rlwrap nc [IP_ADDR] [PORT]`, the `print` function does not work:

```
Python 3.7.3 (default, Dec 20 2019, 18:57:59)
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
---
This console is running in an asyncio event loop.
It allows you to wait for coroutines using the '{0}' syntax.
Try: await asyncio.sleep(1, result=3)
---
>>> print("yes")
>>>
```

I also attempted it without using rlwrap by executing `nc [IP_ADDR] [PORT]` but the behavior stayed the same. I've seen issue #40, but the problem persists here.

Any help? Is it the custom `locals` variable that's giving it trouble?

**Comment by vxgmichel** [comment]
Hi @louie-github, thanks for the report :)

> Any help? Is it the custom locals variable that's giving it trouble?

Good catch! It is definitely a bug related to having the same local dict shared between clients, which is pretty bad. This can easily be fixed though:
```diff
diff --git a/aioconsole/console.py b/aioconsole/console.py
index 8056b7c..b7b7261 100644
--- a/aioconsole/console.py
+++ b/aioconsole/console.py
@@ -61,11 +61,11 @@ class AsynchronousConsole(code.InteractiveConsole):
         self.prompt_control = prompt_control
         self.compile = AsynchronousCompiler()
         # Populate locals
-        self.locals.setdefault("asyncio", asyncio)
-        self.locals.setdefault("loop", self.loop)
-        self.locals.setdefault("print", self.print)
-        self.locals.setdefault("help", self.help)
-        self.locals.setdefault("ainput", self.ainput)
+        self.locals["asyncio"] = asyncio
+        self.locals["loop"] = self.loop
+        self.locals["print"] = self.print
+        self.locals["help"] = self.help
+        self.locals["ainput"] = self.ainput
 
     @functools.wraps(print)
     def print(self, *args, **kwargs):
diff --git a/aioconsole/server.py b/aioconsole/server.py
index 8366fff..ec21a10 100644
--- a/aioconsole/server.py
+++ b/aioconsole/server.py
@@ -48,12 +48,16 @@ async def start_console_server(
     *,
     loop=None
 ):
-    factory = partial(
-        console.AsynchronousConsole,
-        locals=locals,
-        filename=filename,
-        prompt_control=prompt_control,
-    )
+
+    def factory(streams):
+        client_locals = dict(locals) if locals is not None else None
+        return console.AsynchronousConsole(
+            streams=streams,
+            locals=client_locals,
+            filename=filename,
+            prompt_control=prompt_control,
+        )
+
     server = await start_interactive_server(
         factory, host=host, port=port, path=path, banner=banner, loop=loop
     )
```

I'll push this in a PR soon.

**Comment by vxgmichel** [comment]
Fixed in version [v0.2.1](https://github.com/vxgmichel/aioconsole/releases/tag/v0.2.1)!

Thanks again :)

**Comment by louie-github** [comment]
Woah, that was fast. Thank you so much!
